### ISSUE TITLE
/proc/pid/maps reports incorrect dev and inode numbers for files on stackable file systems (e.g. overlayfs)

### ISSUE THREAD

**Issue by avagin** [checkpoint-restore/criu#2312]
As far as I understand this mount related:

```
mnt: Path `/usr/lib64/libc.so.6' resolved to `./' mountpoint"
```

I don't know if and how to correctly fix it, but it seems using a mount namespace in a mount namespace does not work as expected.

The reproducer for me is. This works without a container:
```
setcap cap_checkpoint_restore,cap_sys_ptrace+eip criu/criu
sudo --user=#65534 --group=#65534 unshare -Ucfpm --mount-proc -- bash -c "./test/zdtm.py run -t zdtm/static/maps00 -k always  -f h --rootless && true"
```
But it does not work in a container. @Snorch can you say if this should work and how?

_Originally posted by @adrianreber in https://github.com/checkpoint-restore/criu/pull/2307#discussion_r1413635605_

**Comment by adrianreber** [comment]
@avagin Should I update my PR with the link to this issue and your upstream patch as a comment why the test is disabled during testing in a container or do you want to wait until your change is merged?

**Comment by avagin** [comment]
@adrianreber sorry for the delay. I am on vacation. You can add a comment and merge this pr. In the ideal case, we need to skip this test only if a kernel does have the issue.

**Comment by adrianreber** [comment]
@avagin Merged with an additional comment describing the situation and referring to this issue.

**Comment by rst0git** [comment]
@avagin I was wondering if this problem is also related to https://github.com/checkpoint-restore/criu/issues/136?

**Comment by avagin** [comment]
The kernel patch (https://github.com/torvalds/linux/commit/3efdc78fdc21) was merged and will be in v6.8.
